app/components/request/message.js

```diff
@@ -177,6 +177,46 @@ export default Component.extend({
     return `${format(key)}: skipping stage #${format(args.number)} because its condition does not match: ${format(args.condition)}`;
   },
 
+  invalid_config_format() {
+    return 'invalid config format (must be a hash)';
+  },
+
+  invalid_ref(_, args) {
+    return args.message;
```

Comment:
i understand that ideally the UI should be the party that constructs human readable messages.

the difference between these messages, and the messages above is that these are in response to API error conditions, while the messages above are validation messages that are produced throughout a successful validation process.

while i'd love to return structured data that the UI could then interpret on its own, API v3 does not provide a mechanism for returning such data on error conditions https://developer.travis-ci.com/resource/error#Error, and i think folding such change to API into this one here would extend the scope of this PR unproportionally.

since these are "just" error conditions i think what we have here is good enough. these messages are produced in travis-yml, and they should be rather readable. e.g. the error message format for this one here would be `invalid import reference: %s` with `%s` being the actual import reference provided by the user.

Reply:
I see, @svenfuchs, thanks for the explanation! Let's keep it as is then 👍